lib/internal/stream_base_commons.js

```diff
@@ -213,6 +213,16 @@ function onStreamRead(arrayBuffer) {
     if (stream[kMaybeDestroy])
       stream.on('end', stream[kMaybeDestroy]);
 
+    // TODO(ronag): Without this `readStop`, `onStreamRead`
+    // will be called once more (i.e. after Readable.ended)
+    // on Windows causing a ECONNRESET, failing the
+    // test-https-truncate test.
+    if (handle.readStop) {
+      const err = handle.readStop();
+      if (err)
+        return stream.destroy(errnoException(err, 'read'));
+    }
```

Comment:
@addaleax: Any idea of what's going on here?

Reply:
This behavior occurs only with:

- win10-COMPILED_BY-vs2019
- win2012r2-COMPILED_BY-vs2019-x86

Reply:
@addaleax Any chance you will be able to look into this near term? I've had a VM running for a bit in case you would like me to show you. If you won't have time to look into this near term or don't need me to show you, I'd like to shut down the VM.

Reply:
OK, I'm not that worried about this anymore. It's because we don't `destroy()` in the same tick as `'end'` (see change here, https://github.com/nodejs/node/pull/31806/files#diff-e6ef024c3775d787c38487a6309e491dL636). Requiring a `destroy()` in the same tick I would consider a bug so calling readStop() here as an alternative should be fine.

Reply:
It's not clear why onStreamRead would be called again. Glad that it's not an issue here but it's something worth looking into.

Reply:
https://github.com/nodejs/node/issues/32487